File: nodejs/external-user-service/user-service/src/api/authorization/user.role/user.role.routes.ts
import express from 'express';
import { UserRoleController } from './user.role.controller';
import { auth } from '../../../auth/user.auth/user.auth.handler';
import { UserRoleAuth } from './user.role.auth';

///////////////////////////////////////////////////////////////////////////////////////

export const register = (app: express.Application): void => {

    const router = express.Router();
    const controller = new UserRoleController();

    router.post('/:userId/roles/:roleId', auth(UserRoleAuth.addUserRole), controller.addUserRole);
    router.get('/:userId', auth(UserRoleAuth.getUserRoles), controller.getUserRoles);
    router.delete('/:userId/roles', auth(UserRoleAuth.removeAllUserRoles), controller.removeAllUserRoles);
    router.delete('/:userId/roles/:roleId', auth(UserRoleAuth.removeUserRole), controller.removeUserRole);

    app.use('/api/v1/user-roles', router);
};

File: dotnet/external-user-service/UserService/shala.api/api/users/user.roles/UserRoleRoutes.cs
using FluentValidation;
using Microsoft.AspNetCore.Mvc;
using shala.api.domain.types;

namespace shala.api;

public static class UserRoleRoutes
{
    static readonly string BaseContextName = "Users.UserRoles";

    public static void Map(WebApplication app)
    {

        var router = app.MapGroup("/api/v1/user-roles");

        router.MapPost("/{userId}/roles/{roleId}", async (
            [FromServices] UserRoleController controller,
            Guid userId,
            Guid roleId,
            HttpContext context) => {
            var result = await controller.AddTenantRoleForUser(context, userId, roleId);
            return result;
        })
            .Accepts<Guid>("application/json")
            .WithName($"{BaseContextName}.AddTenantRoleForUser")
            .WithOpenApi()
            .RequireAuthorization()
            .Produces<IResult>(StatusCodes.Status200OK)
            .Produces(StatusCodes.Status400BadRequest)
            .Produces(StatusCodes.Status401Unauthorized)
            .Produces(StatusCodes.Status404NotFound);

        router.MapDelete("/{userId}/roles/{roleId}", async (
            [FromServices] UserRoleController controller,
            Guid userId,
            Guid roleId,
            HttpContext context) => {
            var result = await controller.RemoveTenantRoleForUser(context, userId, roleId);
            return result;
        })
            .Accepts<Guid>("application/json")
            .WithName($"{BaseContextName}.RemoveTenantRoleForUser")
            .WithOpenApi()
            .RequireAuthorization()
            .Produces<IResult>(StatusCodes.Status200OK)
            .Produces(StatusCodes.Status400BadRequest)
            .Produces(StatusCodes.Status401Unauthorized)
            .Produces(StatusCodes.Status404NotFound);

        router.MapGet("/{userId}", async (
            [FromServices] UserRoleController controller,
            Guid userId,
            HttpContext context) => {
            var result = await controller.GetRolesForUser(context, userId);
            return result;
        })
            .Accepts<Guid>("application/json")
            .WithName($"{BaseContextName}.GetRolesForUser")
            .WithOpenApi()
            .RequireAuthorization()
            .Produces<IResult>(StatusCodes.Status200OK)
            .Produces(StatusCodes.Status400BadRequest)
            .Produces(StatusCodes.Status401Unauthorized)
            .Produces(StatusCodes.Status404NotFound);

    }

}
